src/world_model/mdp.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any
+from gymnasium import Env
+
+class MDP(Env):
```

Comment:
Maybe a different structure would be better - `MDP` is just an `abc.ABC`, doesn't necessarily inherit from Env. It should only contain the `state` method. Then it effectively functions as a trait/type class, i.e. when something inherits from `MDP`, it will only check that it contains `state` and that's it.

Reply:
the point here is that an MDP can reset to a particular state, which a normal Env cannot. I have now split this into two classes, `MDP` with only `state()` and `MDPEnv`, which adds `reset(..., state=...)`.